### PR TITLE
List buckets webmachine handler

### PR DESCRIPTION
**Pull request by lukebakken**

### CODE REVIEW

File: include/riak_s3_api.hrl [hunk 1]
```diff
@@ -0,0 +1,6 @@
+-record(list_buckets_response_v1, {
+          user :: rs3_user(),
+          buckets :: [s3_bucket()]
+         }).
+-type list_buckets_response() :: #list_buckets_response_v1{}.
+-define(LBRESP, #list_buckets_response_v1).
```
Comment (lukebakken) [inline include/riak_s3_api.hrl]: @jvoegele @lehoff  this is verbatim from Riak CS. I assume we'd like to hide records in the same manner as what's been going on in `riak_kv`, right?

**Comment by paulhenrich** [comment]
👍

File: include/riak_s3_types.hrl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+-export_type([list_buckets_response/0]).
+
+-record(list_buckets_response_v1, {
+          user :: rs3_user(),
+          buckets :: [s3_bucket()]
+         }).
+-opaque list_buckets_response() :: #list_buckets_response_v1{}.
```
Comment (jvoegele) [inline include/riak_s3_types.hrl]: Should the these types be defined in `riak_s3_types.erl` instead of the header file?

Comment (lukebakken) [inline include/riak_s3_types.hrl]: If they aren't defined here, then I can't reference them [here](https://github.com/basho/riak_s3_api/blob/features/lrb/wm-buckets/src/riak_s3_wm_buckets.erl#L15). I may be doing something wrong, however.

**Comment by lukebakken** [comment]
@jbrisbin @jvoegele all set for review

**Comment by lukebakken** [comment]
I addressed @jvoegele's comment about the types include file, will merge after build passes.

**Pull request merged**